src/main/java/org/codeforamerica/messaging/services/MessageService.java

```diff
@@ -128,6 +128,14 @@ public Optional<Message> getMessage(Long id) {
         return messageRepository.findById(id);
     }
 
+    public Optional<MessageBatch> getMessageBatch(Long id) {
+        Optional<MessageBatch> messageBatch = messageBatchRepository.findById(id);
+        if (messageBatch.isPresent()) {
+            messageBatch.get().setMetrics(messageRepository.getMetrics(id));
```

Comment:
Have you considered limiting the frequency of updates here/caching? Not important currently, but might it be later?

Reply:
Can you elaborate?

Reply:
How frequently might someone be hitting `getMessageBatch`, trying to get real-ish time updates on statuses? I think it will be a long time before we have the number of batches and users where this would cause strain, but hypothetically, would this be an appropriate place to add some logic so that not every call will automatically update the metrics? I might be vastly over cautious here so just wondering.

Reply:
If someone is hitting the GET endpoint to get real-ish time updates, they would be better served by getting true values rather than cached ones. I also think that the query would be performant if we add an index on `message.message_batch_id`.